backend/src/modules/idim-webservice/idim-webservice.service.ts

```diff
@@ -0,0 +1,131 @@
+    private idimWebServiceUsername = process.env.IDIM_WEB_SERVICE_USERNAME;
+    private idimWebServicePassword = process.env.IDIM_WEB_SERVICE_PASSWORD;
+
+    async verifyIdirUser(
+        userId: string,
+        requesterUserId: string,
+        requesterAccountTypeCode: string
+    ): Promise<HttpException | IDIRUserResponse> {
+        if (
+            !this.idimWebServiceUrl ||
+            !this.idimWebServiceID ||
+            !this.idimWebServiceUsername ||
+            !this.idimWebServicePassword
+        ) {
+            return new HttpException(
+                {
+                    error: 'Missing IDIM web service crednetials in the environment variables',
+                },
+                HttpStatus.BAD_REQUEST
+            );
+        }
+
+        // add autorization header for making the soap api call
+        const auth =
+            'Basic ' +
+            Buffer.from(
+                this.idimWebServiceUsername + ':' + this.idimWebServicePassword
+            ).toString('base64');
+
+        try {
+            const client = await soap.createClientAsync(
+                this.idimWebServiceUrl,
+                {
+                    wsdl_headers: { Authorization: auth },
+                }
+            );
+            client.addHttpHeader('Authorization', auth);
+
+            // set xml schema parameters for the request
+            const requestData = {
+                internalAccountSearchRequest: {
+                    onlineServiceId: this.idimWebServiceID,
+                    // who is sending the request
+                    requesterAccountTypeCode,
+                    requesterUserId,
+                    // some config parameter
+                    pagination: { pageSizeMaximum: '20', pageIndex: '1' },
+                    sort: { direction: 'Ascending', onProperty: 'UserId' },
+                    // who we search for, exact match userID
+                    accountMatch: {
+                        userId: {
+                            value: userId,
+                            matchPropertyUsing: 'Exact',
+                        },
+                    },
+                },
+            };
+
+            return new Promise((resolve, reject) => {
+                client.BCeIDService.BCeIDServiceSoap.searchInternalAccount(
+                    requestData,
+                    function (error, foundUser) {
+                        if (error) {
+                            reject(
+                                new HttpException(
+                                    {
+                                        error:
+                                            'IDIM web service call error: ' +
+                                            error,
+                                    },
+                                    HttpStatus.INTERNAL_SERVER_ERROR
+                                )
+                            );
+
+                            return null;
```

Comment:
Since it is being reject(), no need to return a "null" I think, maybe try if that is correct.